RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,282 @@
+
+// Package:    RecoMuon_TrackerSeedGenerator
+// Class:      MuonHLTSeedMVAClassifier
+
+// Original Author:  Won Jun, OH Minseok
+//         Created:  Fri, 28 May 2021
+
+// system include files
+#include <memory>
+#include <cmath>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+// Geometry
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// TrajectorySeed
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeed.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/TrajectorySeed/interface/PropagationDirection.h"
+#include "DataFormats/TrajectoryState/interface/PTrajectoryStateOnDet.h"
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
+
+#include "RecoMuon/TrackerSeedGenerator/interface/SeedMvaEstimator.h"
+
+// class declaration
+bool sortByMvaScore(const std::pair<unsigned, double>& A, const std::pair<unsigned, double>& B) {
+  return (A.second > B.second);
+};
+
+class MuonHLTSeedMVAClassifier : public edm::stream::EDProducer<> {
+public:
+  explicit MuonHLTSeedMVAClassifier(const edm::ParameterSet&);
+  ~MuonHLTSeedMVAClassifier() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  // member data
+  const edm::EDGetTokenT<TrajectorySeedCollection> seedToken_;
+  const edm::EDGetTokenT<l1t::MuonBxCollection> l1MuonToken_;
+  const edm::EDGetTokenT<reco::RecoChargedCandidateCollection> l2MuonToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> trackerGeometryToken_;
+
+  typedef std::pair<std::unique_ptr<const SeedMvaEstimator>, std::unique_ptr<const SeedMvaEstimator>>
+      pairSeedMvaEstimator;
```

Comment:
```suggestion
  typedef std::pair<std::unique_ptr<const SeedMvaEstimator>, std::unique_ptr<const SeedMvaEstimator>>
      PairSeedMvaEstimator;
```
would be more along the naming convention.